Split/Engine/DefaultTreatmentManager.swift

```diff
@@ -0,0 +1,158 @@
+            }
+        }
+        
+        if splits.count > 0 {
+            let splitsNoDuplicated = Set(splits.filter { !$0.isEmpty() }.map { $0 })
+            for splitName in splitsNoDuplicated {
+                results[splitName] = getTreatmentWithConfigNoMetrics(splitName: splitName, shouldValidate: false, attributes: attributes, validationTag: validationTag)
+            }
+        } else {
+            Logger.d("\(validationTag): split_names is an empty array or has null values")
+        }
+        return results
+    }
+    
+    private func getTreatmentWithConfigNoMetrics(splitName: String, shouldValidate: Bool = true, attributes:[String:Any]? = nil, validationTag: String) -> SplitResult {
+        
+        if shouldValidate {
+            if let errorInfo = keyValidator.validate(matchingKey: key.matchingKey, bucketingKey: key.bucketingKey) {
+                validationLogger.log(errorInfo: errorInfo, tag: validationTag)
+                return SplitResult(treatment: SplitConstants.CONTROL)
+            }
+        }
+        
+        if let errorInfo = splitValidator.validate(name: splitName) {
+            validationLogger.log(errorInfo: errorInfo, tag: validationTag)
+            if errorInfo.isError {
+                return SplitResult(treatment: SplitConstants.CONTROL)
+            }
+        }
+        
+        if let errorInfo = splitValidator.validateSplit(name: splitName) {
+            validationLogger.log(errorInfo: errorInfo, tag: validationTag)
+            if errorInfo.isError || errorInfo.hasWarning(.nonExistingSplit) {
+                return SplitResult(treatment: SplitConstants.CONTROL)
+            }
+        }
+        
+        let trimmedSplitName = splitName.trimmingCharacters(in: .whitespacesAndNewlines)
+        
+        do {
+            let result = try evaluateIfReady(splitName: trimmedSplitName, attributes: attributes)
+            logImpression(label: result.label, changeNumber: result.splitVersion, treatment: result.treatment, splitName: trimmedSplitName, attributes: attributes)
+            return SplitResult(treatment: result.treatment, config: result.configuration)
+        }
+        catch {
+            logImpression(label: ImpressionsConstants.EXCEPTION, treatment: SplitConstants.CONTROL, splitName: trimmedSplitName, attributes: attributes)
+            return SplitResult(treatment: SplitConstants.CONTROL)
+        }
+    }
+    
+    private func evaluateIfReady(splitName: String, attributes:[String:Any]?) throws -> EvaluationResult {
+        if !isSdkReady() {
+            return EvaluationResult(treatment: SplitConstants.CONTROL, label: ImpressionsConstants.NOT_READY);
+        }
+        return try evaluator.evalTreatment(matchingKey: key.matchingKey, bucketingKey: key.bucketingKey, splitName: splitName, attributes: attributes)
+    }
+    
+    private func logImpression(label: String, changeNumber: Int64? = nil, treatment: String, splitName: String, attributes:[String:Any]? = nil) {
```

Comment:
if some errors occurs, will this stop the execution of the entire program? if that is the case, we should add some catching here to avoid it